### PR TITLE
[WFCORE-5079] Refactoring of "management" layers.

### PR DESCRIPTION
**Pull request by darranl**
https://issues.redhat.com/browse/WFCORE-5079
https://issues.redhat.com/browse/WFCORE-5084

Switches to two secured layers only, this will make it impossible to accidentally provision an unsecured server.

This also contains a small correction to the Elytron layer which was bringing in management specific RBAC configuration.

The following diff illustrates the effect of adding the elytron layer to base-server:
https://gist.github.com/darranl/0447770f0a7a35998225277caa2b4549/revisions#diff-d07c1929437497a0dbbef91e2c232b3c

The next diff illustrates the result of the management layer:
https://gist.github.com/darranl/0447770f0a7a35998225277caa2b4549/revisions#diff-c3a1460cbad7b8a1f5176b566a2ac35a

Finally the legacy-management layer can be visualised here:
https://gist.github.com/darranl/0447770f0a7a35998225277caa2b4549/revisions#diff-6df06b002fa7b64133a9cbe69cc1aa48

A final commit is to add back a "secure-management" layer, this is just an alias to "management" until WildFly can move over to use "management" instead.

### CODE REVIEW

**Comment by darranl** [comment]
@yersan / @jmesnil My local testing has taken me to the point of testing the resulting build of WildFly with these change in place, is there anything else beyond that in the area of S2I that may still need some consideration?

**Comment by yersan** [comment]
@darranl I'm unable to point out to you to a specific consideration. I can give you some context about how it is working on S2I though. 

The Galleon pack used in S2I is built upon the Galleon pack defined in WildFly, it is a [user defined feature pack](https://docs.wildfly.org/galleon-plugins/#_build_user_feature_pack). All the layers defined in WildFly are inherited in S2I. The S2I Galleon pack redefines some of them tweaking the feature configurations for the OpenShift context. Behind scenes, Galleon merges the configurations done in S2I and the ones inherited from WildFly.

My understanding is we have to take a closer look at the layers defined in S2I upon these changes, but I'm assuming that it's a task to be done in the S2I context (new Jiras maybe?) when the S2I Galleon pack changes and starts inheriting from a newer Galleon pack version released on WildFly.

One think we can do is review the layers definitions in the S2I context as they are now and figure out what can be affected, does it make sense to do it now?

**Comment by darranl** [comment]
A quick review of S2I layers may be useful, just to double check this is not going to cause any surprises for anyone then at least we know if there is follow up.

File: core-feature-pack/galleon-feature-pack/src/main/resources/configs/standalone/standalone.xml/config.xml [hunk 1]
```diff
@@ -3,6 +3,8 @@
 <config xmlns="urn:jboss:galleon:config:1.0" name="standalone.xml" model="standalone">
     <layers>
         <include name="legacy-management"/>
+        <exclude name="management"/> <!-- Other layers may optionally depend on this one. -->
```
[the remark this answers is not in the document]
Comment (darranl) [reply core-feature-pack/galleon-feature-pack/src/main/resources/configs/standalone/standalone.xml/config.xml:6]: Management interfaces don't use the legacy security subsystem, they use the legacy security realms which are a part of WildFly Core.  The inclusion of "legacy-management" on the prior line still results in secured management interfaces in the configuration.

**Pull request merged**